hathor/transaction/base_transaction.py

```diff
@@ -525,9 +525,8 @@ def validate_full(self, skip_block_weight_verification: bool = False, sync_check
             soft_height = meta.soft_height
             meta.soft_height = None
             meta.height = self.calculate_height()
-            if soft_height is not None and soft_height != meta.height:
-                self.log.error('A wrong height was previously set as soft height, this indicates a but',
-                               height=meta.height, soft_height=soft_height)
+            if soft_height is not None:
+                assert soft_height == meta.height, 'A wrong height was previously set as soft height. Sync bug?'
```

Comment:
Where is this method called from? I'm not sure whether it should be an assert or another verification.

Reply:
This method is basically called from `HathorManager.on_new_tx`. In theory `soft_height` can only be wrong if there is a wrong checkpoint configured. I used an assert because I wanted to check that that is in fact the case, just so we don't update the soft_height with a calculated height of different value.